### PR TITLE
[AzureStack][Admin API] Adding azurestack AKS admin spec

### PR DESCRIPTION
**Pull request by Bhuvaneswari-Santharam**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [x] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [x] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/azsadmin/resource-manager/containerservice/Microsoft.ContainerService.Admin/preview/2019-11-01/examples/Operations/List.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
+{
+  "parameters": {
```
Comment (bganapa) [inline specification/azsadmin/resource-manager/containerservice/Microsoft.ContainerService.Admin/preview/2019-11-01/examples/Operations/List.json:2]: folder name should be 2019-11-01-preview

**Review by bganapa — APPROVED** [review]

**Comment by leni-msft** [comment]
@bganapa please fix the lint warnings, per ARM requirements

**Comment by Bhuvaneswari-Santharam** [comment]
@leni-msft - All the lint warning are fixed. Could you please review?

**Comment by chiragg4u** [comment]
Please ensure you've reviewed following guidelines including ARM resource provider contract and REST guidelines. Estimated time (4 hours). This is required before you can request review from ARM API Review board.

**Comment by chiragg4u** [comment]
Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.

Please create new branch with this recommendation for review to be continued.

**Comment by bganapa** [comment]
@chiragg4u This is a new RP spec. There is no existing API spec for this RP. The RP is specific to Azure Stack Operator and wont be deployed in azure

**Comment by jianyexi** [comment]
@chiragg4u could you review the PR

File: specification/azsadmin/resource-manager/containerservice/Microsoft.ContainerService.Admin/preview/2019-11-01-preview/managedClusters.json [hunk 2]
```diff
@@ -0,0 +1,153 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2019-11-01",
```
Comment (bganapa) [inline specification/azsadmin/resource-manager/containerservice/Microsoft.ContainerService.Admin/preview/2019-11-01-preview/managedClusters.json:4]: please confirm whether you want the apiversion to be preview or not

File: specification/azsadmin/resource-manager/containerservice/Microsoft.ContainerService.Admin/preview/2019-11-01-preview/managedClusters.json [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+        "operationId": "ContainerServices_List",
+        "parameters": [
+          {
+            "$ref": "quotas.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "quotas.json#/parameters/LocationParameter"
+          },
+          {
+            "$ref": "quotas.json#/parameters/ApiVersionParameter"
```
Comment (bganapa) [inline specification/azsadmin/resource-manager/containerservice/Microsoft.ContainerService.Admin/preview/2019-11-01-preview/managedClusters.json:39]: pls use common-types wherever possible

File: specification/azsadmin/resource-manager/containerservice/Microsoft.ContainerService.Admin/preview/2019-11-01-preview/examples/Operations/List.json [hunk 4]
```diff
@@ -0,0 +1,40 @@
+{
+  "parameters": {
+    "api-version": "2019-11-01-preview"
```
Comment (bganapa) [inline specification/azsadmin/resource-manager/containerservice/Microsoft.ContainerService.Admin/preview/2019-11-01-preview/examples/Operations/List.json:3]: if thsi api is not preview, pls change the folder name to stable

**Comment by zhenglaizhang** [comment]
Default branch of the spec repo has been updated as main, I have updated the PR to target main branch

**Comment by bganapa** [comment]
@jianyexi Could you pls help review this PR and merge?

**Comment by jianyexi** [comment]
@chiragg4u  could you review it ?

**Comment by bganapa** [comment]
@chiragg4u This is a new API Spec. The API surface is exposed only for AzureStack Operator (differentiated with .Admin in the namespace).  these are not exposed in Azure. Appreciate your review and approval on this PR as we are looking to ship the powershell interface. The PR earlier got closed by bot for inactivity.

**Comment by bganapa** [comment]
@chiragg4u Could you pls review?

File: specification/azsadmin/resource-manager/containerservice/Microsoft.ContainerService.Admin/stable/2019-11-01/managedClusters.json [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+        }
+      }
+    }
+  },
+  "definitions": {
+    "ContainerServiceProperties": {
+      "description": "Container service properties.",
+      "type": "object",
+      "properties": {
+        "id": {
```
Comment (pilor) [inline specification/azsadmin/resource-manager/containerservice/Microsoft.ContainerService.Admin/stable/2019-11-01/managedClusters.json:67]: id, name, location, and subscriptionId are either top level properties that must not be repeated inside the property bag or implied by the URL in subscriptionIDs case.  Please remove them from the property bag.  https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#resource-request-properties-envelope

File: specification/azsadmin/resource-manager/containerservice/Microsoft.ContainerService.Admin/stable/2019-11-01/managedClusters.json [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+          "description": "Container service location.",
+          "type": "string",
+          "readOnly": true
+        },
+        "subscriptionId": {
+          "description": "SubscriptionId where container service is present.",
+          "type": "string",
+          "readOnly": true
+        },
+        "creationDate": {
```
Comment (pilor) [inline specification/azsadmin/resource-manager/containerservice/Microsoft.ContainerService.Admin/stable/2019-11-01/managedClusters.json:87]: if creation/lastmodifeddate info is necessary it must utilize the systemData contract:  https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/common-api-contracts.md#system-metadata-for-all-azure-resources

File: specification/azsadmin/resource-manager/containerservice/Microsoft.ContainerService.Admin/stable/2019-11-01/managedClusters.json (hunk 5 above)
Comment (Bhuvaneswari-Santharam) [reply specification/azsadmin/resource-manager/containerservice/Microsoft.ContainerService.Admin/stable/2019-11-01/managedClusters.json:67]: ID, subscriptionId, location under properties are Azure Stack user namespace kubernetes cluster details. These are different from Azure Stack Admin namespace id, subscriptionId and location top level entities

File: specification/azsadmin/resource-manager/containerservice/Microsoft.ContainerService.Admin/stable/2019-11-01/managedClusters.json (hunk 6 above)
Comment (Bhuvaneswari-Santharam) [reply specification/azsadmin/resource-manager/containerservice/Microsoft.ContainerService.Admin/stable/2019-11-01/managedClusters.json:87]: Its a location based proxy only resource, not a tracked top level resource

**Review by bganapa — APPROVED** [review]

**Comment by bganapa** [comment]
@jianyexi  could you pls merge this PR?

**Pull request merged**